integrations/optimum/src/haystack_integrations/components/embedders/optimum/_backend.py

```diff
@@ -188,6 +188,12 @@ def pool_embeddings(self, model_output: torch.Tensor, attention_mask: torch.Tens
         pooled_outputs = self.pooling_layer.forward(features)
         return pooled_outputs["sentence_embedding"]
 
+    @overload
```

Comment:
interesting, did not know about this `@overload` decorator.

I notice that we are using the decorator here, but below we are still keeping the typing with the Union - do need to keep both, or we can remove the one below?

Reply:
ok